src/ya-comiste-react-native/src/BlockRenderer.js

```diff
@@ -0,0 +1,80 @@
+// @flow
+
+import * as React from 'react';
+import { Text } from 'react-native';
+import { createFragmentContainer, graphql } from 'react-relay';
+import { warning } from '@adeira/js';
+
+import type { FragmentContainerType } from './relay/relayTypes';
+import type { BlockRendererFragment } from './__generated__/BlockRendererFragment.graphql';
+
+type Props = {|
+  +block: BlockRendererFragment,
+  +componentId: string,
+|};
+
+function BlockRenderer(props) {
+  const block = props.block.block;
+  let Component = null;
+
+  // TODO: `@match` & `@module` 3D rendering (?)
```

Comment:
I don't think that would work for react-native, since all assets have to be in the bundle.

Reply:
That is true, I was thinking the same. But the concept is still valid and you can use the same techniques. The only difference is that you are not going to be lazy loading the components over the network, no? You can still leverage the automatic querying (with the automatic `supported` arg propagation). I see it quite beneficial even in RN world.

Reply:
> You can still leverage the automatic querying (with the automatic supported arg propagation). I see it quite beneficial even in RN world.

Yeah, I didn't think about this, but I did a quick googling, and it seems using react.lazy should speed up start-up time for the apps so 👍